libazureinit/src/config.rs

```diff
@@ -421,6 +492,42 @@ impl Config {
     }
 }
 
+/// Checks that the first line of the file is `# ... azure-init-config`
+/// allowing any amount of whitespace between `#` and `azure-init-config`.`
+/// Returns an error if the file doesn't start with `#azure-init-config`.
+fn check_azure_init_header(file_path: &Path) -> Result<(), Error> {
+    if !file_path.is_file() {
+        // If it's not a file, ignore or skip the check
+        return Ok(());
+    }
+
+    let file = fs::File::open(file_path).map_err(|e| {
+        tracing::error!("Failed to open file {:?}: {:?}", file_path, e);
+        Error::Io(e)
+    })?;
+    let mut reader = std::io::BufReader::new(file);
+    let mut first_line = String::new();
+    let _ = reader.read_line(&mut first_line).map_err(|e| {
+        tracing::error!("Error reading first line of {:?}: {:?}", file_path, e);
+        Error::Io(e)
+    })?;
+
+    let trimmed = first_line.trim();
+    let re = Regex::new(r"^#\s*azure-init-config$").unwrap();
+    if !re.is_match(trimmed) {
+        tracing::error!(
```

Comment:
This should not be an error but just an info. user-data can be used for many purposes and the customers don't necessarily want azure-init to handle it.